File: backend/models/car.go
package models

type CarBrand struct {
	ID      int     `json:"id"`
	Name    string  `json:"name"`
	NameTh  *string `json:"name_th"`
	Country *string `json:"country"`
	LogoURL *string `json:"logo_url"`
}

type CarBrandWithModels struct {
	CarBrand
	Models []CarModel `json:"models"`
}

type CarModel struct {
	ID             int    `json:"id"`
	BrandID        int    `json:"brand_id"`
	Name           string `json:"name"`
	PowertrainType string `json:"powertrain_type"`
	BodyType       string `json:"body_type"`
	Segment        string `json:"segment"`
	YearLaunched   *int   `json:"year_launched"`
	Status         string `json:"status"`
	// Joined fields
	BrandName *string `json:"brand_name,omitempty"`
}

type CarModelWithVariants struct {
	CarModel
	Variants []CarVariantSummary `json:"variants"`
}

type CarVariantSummary struct {
	ID        int      `json:"id"`
	ModelID   int      `json:"model_id"`
	Name      string   `json:"name"`
	PriceBaht *float64 `json:"price_baht"`
	Status    string   `json:"status"`
}

type CarVariant struct {
	ID        int      `json:"id"`
	ModelID   int      `json:"model_id"`
	Name      string   `json:"name"`
	PriceBaht *float64 `json:"price_baht"`
	Status    string   `json:"status"`

	// Joined fields
	BrandName      *string `json:"brand_name,omitempty"`
	ModelName      *string `json:"model_name,omitempty"`
	PowertrainType *string `json:"powertrain_type,omitempty"`
	BodyType       *string `json:"body_type,omitempty"`

	// Electric / Battery
	BatteryCapacityKwh    *float64 `json:"battery_capacity_kwh"`
	BatteryType           *string  `json:"battery_type"`
	MotorPowerKw          *float64 `json:"motor_power_kw"`
	MotorTorqueNm         *float64 `json:"motor_torque_nm"`
	FrontMotorKw          *float64 `json:"front_motor_kw"`
	RearMotorKw           *float64 `json:"rear_motor_kw"`
	RangeKm               *int     `json:"range_km"`
	RangeStandard         *string  `json:"range_standard"`
	AcChargeKw            *float64 `json:"ac_charge_kw"`
	DcChargeKw            *float64 `json:"dc_charge_kw"`
	AcChargeTimeHrs       *float64 `json:"ac_charge_time_hrs"`
	DcChargeTimeMins      *int     `json:"dc_charge_time_mins"`
	ChargingPort          *string  `json:"charging_port"`
	V2l                   *bool    `json:"v2l"`
	V2g                   *bool    `json:"v2g"`
	HeatPump              *bool    `json:"heat_pump"`
	BatteryPreconditioning *bool   `json:"battery_preconditioning"`

	// Engine (ICE / Hybrid)
	DisplacementCc     *int     `json:"displacement_cc"`
	EngineType         *string  `json:"engine_type"`
	Horsepower         *int     `json:"horsepower"`
	EngineTorqueNm     *int     `json:"engine_torque_nm"`
	FuelType           *string  `json:"fuel_type"`
	FuelTankLiters     *float64 `json:"fuel_tank_liters"`
	FuelConsumptionKml *float64 `json:"fuel_consumption_kml"`
	Turbo              *bool    `json:"turbo"`
	Transmission       *string  `json:"transmission"`
	TransmissionSpeeds *int     `json:"transmission_speeds"`

	// Combined (Hybrid)
	SystemPowerHp  *int `json:"system_power_hp"`
	SystemTorqueNm *int `json:"system_torque_nm"`
	EvRangeKm      *int `json:"ev_range_km"`

	// Performance
	TopSpeedKmh      *int     `json:"top_speed_kmh"`
	Acceleration0100 *float64 `json:"acceleration_0_100"`

	// Dimensions
	LengthMm            *int `json:"length_mm"`
	WidthMm              *int `json:"width_mm"`
	HeightMm             *int `json:"height_mm"`
	WheelbaseMm          *int `json:"wheelbase_mm"`
	GroundClearanceMm    *int `json:"ground_clearance_mm"`
	CurbWeightKg         *int `json:"curb_weight_kg"`
	GrossWeightKg        *int `json:"gross_weight_kg"`
	TrunkCapacityLiters  *int `json:"trunk_capacity_liters"`
	TrunkMaxLiters       *int `json:"trunk_max_liters"`
	FrunkCapacityLiters  *int `json:"frunk_capacity_liters"`

	// Drive
	DriveType       *string `json:"drive_type"`
	FrontSuspension *string `json:"front_suspension"`
	RearSuspension  *string `json:"rear_suspension"`
	FrontBrakes     *string `json:"front_brakes"`
	RearBrakes      *string `json:"rear_brakes"`
	TireSizeFront   *string `json:"tire_size_front"`
	TireSizeRear    *string `json:"tire_size_rear"`
	SpareTire       *string `json:"spare_tire"`

	// Safety
	Airbags            *int  `json:"airbags"`
	Abs                *bool `json:"abs"`
	Esc                *bool `json:"esc"`
	TractionControl    *bool `json:"traction_control"`
	HillStartAssist    *bool `json:"hill_start_assist"`
	HillDescentControl *bool `json:"hill_descent_control"`
	Tpms               *bool `json:"tpms"`
	Isofix             *bool `json:"isofix"`
	ParkingSensorFront *bool `json:"parking_sensor_front"`
	ParkingSensorRear  *bool `json:"parking_sensor_rear"`
	CameraRear         *bool `json:"camera_rear"`
	Camera360          *bool `json:"camera_360"`
	AutoParking        *bool `json:"auto_parking"`

	// ADAS
	Aeb                      *bool   `json:"aeb"`
	Fcw                      *bool   `json:"fcw"`
	Lka                      *bool   `json:"lka"`
	Ldw                      *bool   `json:"ldw"`
	Bsd                      *bool   `json:"bsd"`
	Rcta                     *bool   `json:"rcta"`
	Acc                      *bool   `json:"acc"`
	AccStopGo                *bool   `json:"acc_stop_go"`
	DriverMonitoring         *string `json:"driver_monitoring"`
	TrafficSignRecognition   *bool   `json:"traffic_sign_recognition"`
	NightVision              *bool   `json:"night_vision"`
	AdasLevel                *string `json:"adas_level"`

	// NCAP
	NcapRating *float64 `json:"ncap_rating"`
	NcapBody   *string  `json:"ncap_body"`
	NcapYear   *int     `json:"ncap_year"`

	// Comfort
	Seats                *int    `json:"seats"`
	SeatMaterial         *string `json:"seat_material"`
	DriverSeatElectric   *bool   `json:"driver_seat_electric"`
	PassengerSeatElectric *bool  `json:"passenger_seat_electric"`
	DriverSeatMemory     *bool   `json:"driver_seat_memory"`
	VentilatedSeatsFront *bool   `json:"ventilated_seats_front"`
	VentilatedSeatsRear  *bool   `json:"ventilated_seats_rear"`
	HeatedSeatsFront     *bool   `json:"heated_seats_front"`
	HeatedSeatsRear      *bool   `json:"heated_seats_rear"`
	RearSeatRecline      *bool   `json:"rear_seat_recline"`
	AcZones              *int    `json:"ac_zones"`
	RearAcVents          *bool   `json:"rear_ac_vents"`

	// Infotainment
	ScreenSizeInch       *float64 `json:"screen_size_inch"`
	ScreenType           *string  `json:"screen_type"`
	DigitalCluster       *bool    `json:"digital_cluster"`
	ClusterSizeInch      *float64 `json:"cluster_size_inch"`
	Hud                  *bool    `json:"hud"`
	SpeakerBrand         *string  `json:"speaker_brand"`
	SpeakerCount         *int     `json:"speaker_count"`
	AppleCarplay         *bool    `json:"apple_carplay"`
	AndroidAuto          *bool    `json:"android_auto"`
	WirelessCarplay      *bool    `json:"wireless_carplay"`
	WirelessAndroidAuto  *bool    `json:"wireless_android_auto"`
	WirelessPhoneCharging *bool   `json:"wireless_phone_charging"`
	UsbCPorts            *int     `json:"usb_c_ports"`
	UsbAPorts            *int     `json:"usb_a_ports"`
	Bluetooth            *string  `json:"bluetooth"`
	OtaUpdate            *bool    `json:"ota_update"`

	// Exterior
	HeadlightType      *string `json:"headlight_type"`
	Drl                *bool   `json:"drl"`
	AutoHeadlights     *bool   `json:"auto_headlights"`
	AdaptiveHeadlights *bool   `json:"adaptive_headlights"`
	FogLights          *bool   `json:"fog_lights"`
	Sunroof            *string `json:"sunroof"`
	PowerTailgate      *bool   `json:"power_tailgate"`
	HandsFreeTailgate  *bool   `json:"hands_free_tailgate"`
	KeylessEntry       *bool   `json:"keyless_entry"`
	PushStart          *bool   `json:"push_start"`
	AutoFoldingMirrors *bool   `json:"auto_folding_mirrors"`
	RainSensingWipers  *bool   `json:"rain_sensing_wipers"`
	RoofRails          *bool   `json:"roof_rails"`

	// Warranty
	WarrantyYears       *int `json:"warranty_years"`
	WarrantyKm          *int `json:"warranty_km"`
	BatteryWarrantyYears *int `json:"battery_warranty_years"`
	BatteryWarrantyKm   *int `json:"battery_warranty_km"`
}

File: frontend/lib/models/car.dart
class CarBrand {
  final int id;
  final String name;
  final String? nameTh;
  final String? country;
  final String? logoUrl;

  CarBrand({
    required this.id,
    required this.name,
    this.nameTh,
    this.country,
    this.logoUrl,
  });

  factory CarBrand.fromJson(Map<String, dynamic> json) {
    return CarBrand(
      id: json['id'],
      name: json['name'],
      nameTh: json['name_th'],
      country: json['country'],
      logoUrl: json['logo_url'],
    );
  }
}

class CarModel {
  final int id;
  final int brandId;
  final String name;
  final String powertrainType;
  final String bodyType;
  final String segment;
  final int? yearLaunched;
  final String status;
  final String? brandName;

  CarModel({
    required this.id,
    required this.brandId,
    required this.name,
    required this.powertrainType,
    required this.bodyType,
    required this.segment,
    this.yearLaunched,
    required this.status,
    this.brandName,
  });

  factory CarModel.fromJson(Map<String, dynamic> json) {
    return CarModel(
      id: json['id'],
      brandId: json['brand_id'],
      name: json['name'],
      powertrainType: json['powertrain_type'],
      bodyType: json['body_type'],
      segment: json['segment'],
      yearLaunched: json['year_launched'],
      status: json['status'],
      brandName: json['brand_name'],
    );
  }
}

class CarVariantSummary {
  final int id;
  final int modelId;
  final String name;
  final double? priceBaht;
  final String status;

  CarVariantSummary({
    required this.id,
    required this.modelId,
    required this.name,
    this.priceBaht,
    required this.status,
  });

  factory CarVariantSummary.fromJson(Map<String, dynamic> json) {
    return CarVariantSummary(
      id: json['id'],
      modelId: json['model_id'],
      name: json['name'],
      priceBaht: (json['price_baht'] as num?)?.toDouble(),
      status: json['status'],
    );
  }
}

class CarVariant {
  // Basic Info
  final int id;
  final int modelId;
  final String name;
  final double? priceBaht;
  final String status;

  // Joined fields
  final String? brandName;
  final String? modelName;
  final String? powertrainType;
  final String? bodyType;

  // Electric / Battery
  final double? batteryCapacityKwh;
  final String? batteryType;
  final double? motorPowerKw;
  final double? motorTorqueNm;
  final double? frontMotorKw;
  final double? rearMotorKw;
  final int? rangeKm;
  final String? rangeStandard;
  final double? acChargeKw;
  final double? dcChargeKw;
  final double? acChargeTimeHrs;
  final int? dcChargeTimeMins;
  final String? chargingPort;
  final bool? v2l;
  final bool? v2g;
  final bool? heatPump;
  final bool? batteryPreconditioning;

  // Engine (ICE / Hybrid)
  final int? displacementCc;
  final String? engineType;
  final int? horsepower;
  final int? engineTorqueNm;
  final String? fuelType;
  final double? fuelTankLiters;
  final double? fuelConsumptionKml;
  final bool? turbo;
  final String? transmission;
  final int? transmissionSpeeds;

  // Combined (Hybrid)
  final int? systemPowerHp;
  final int? systemTorqueNm;
  final int? evRangeKm;

  // Performance
  final int? topSpeedKmh;
  final double? acceleration0100;

  // Dimensions
  final int? lengthMm;
  final int? widthMm;
  final int? heightMm;
  final int? wheelbaseMm;
  final int? groundClearanceMm;
  final int? curbWeightKg;
  final int? grossWeightKg;
  final int? trunkCapacityLiters;
  final int? trunkMaxLiters;
  final int? frunkCapacityLiters;

  // Drive
  final String? driveType;
  final String? frontSuspension;
  final String? rearSuspension;
  final String? frontBrakes;
  final String? rearBrakes;
  final String? tireSizeFront;
  final String? tireSizeRear;
  final String? spareTire;

  // Safety
  final int? airbags;
  final bool? abs;
  final bool? esc;
  final bool? tractionControl;
  final bool? hillStartAssist;
  final bool? hillDescentControl;
  final bool? tpms;
  final bool? isofix;
  final bool? parkingSensorFront;
  final bool? parkingSensorRear;
  final bool? cameraRear;
  final bool? camera360;
  final bool? autoParking;

  // ADAS
  final bool? aeb;
  final bool? fcw;
  final bool? lka;
  final bool? ldw;
  final bool? bsd;
  final bool? rcta;
  final bool? acc;
  final bool? accStopGo;
  final String? driverMonitoring;
  final bool? trafficSignRecognition;
  final bool? nightVision;
  final String? adasLevel;

  // NCAP
  final double? ncapRating;
  final String? ncapBody;
  final int? ncapYear;

  // Comfort
  final int? seats;
  final String? seatMaterial;
  final bool? driverSeatElectric;
  final bool? passengerSeatElectric;
  final bool? driverSeatMemory;
  final bool? ventilatedSeatsFront;
  final bool? ventilatedSeatsRear;
  final bool? heatedSeatsFront;
  final bool? heatedSeatsRear;
  final bool? rearSeatRecline;
  final int? acZones;
  final bool? rearAcVents;

  // Infotainment
  final double? screenSizeInch;
  final String? screenType;
  final bool? digitalCluster;
  final double? clusterSizeInch;
  final bool? hud;
  final String? speakerBrand;
  final int? speakerCount;
  final bool? appleCarplay;
  final bool? androidAuto;
  final bool? wirelessCarplay;
  final bool? wirelessAndroidAuto;
  final bool? wirelessPhoneCharging;
  final int? usbCPorts;
  final int? usbAPorts;
  final String? bluetooth;
  final bool? otaUpdate;

  // Exterior
  final String? headlightType;
  final bool? drl;
  final bool? autoHeadlights;
  final bool? adaptiveHeadlights;
  final bool? fogLights;
  final String? sunroof;
  final bool? powerTailgate;
  final bool? handsFreeTailgate;
  final bool? keylessEntry;
  final bool? pushStart;
  final bool? autoFoldingMirrors;
  final bool? rainSensingWipers;
  final bool? roofRails;

  // Warranty
  final int? warrantyYears;
  final int? warrantyKm;
  final int? batteryWarrantyYears;
  final int? batteryWarrantyKm;

  CarVariant({
    required this.id,
    required this.modelId,
    required this.name,
    this.priceBaht,
    required this.status,
    this.brandName,
    this.modelName,
    this.powertrainType,
    this.bodyType,
    this.batteryCapacityKwh,
    this.batteryType,
    this.motorPowerKw,
    this.motorTorqueNm,
    this.frontMotorKw,
    this.rearMotorKw,
    this.rangeKm,
    this.rangeStandard,
    this.acChargeKw,
    this.dcChargeKw,
    this.acChargeTimeHrs,
    this.dcChargeTimeMins,
    this.chargingPort,
    this.v2l,
    this.v2g,
    this.heatPump,
    this.batteryPreconditioning,
    this.displacementCc,
    this.engineType,
    this.horsepower,
    this.engineTorqueNm,
    this.fuelType,
    this.fuelTankLiters,
    this.fuelConsumptionKml,
    this.turbo,
    this.transmission,
    this.transmissionSpeeds,
    this.systemPowerHp,
    this.systemTorqueNm,
    this.evRangeKm,
    this.topSpeedKmh,
    this.acceleration0100,
    this.lengthMm,
    this.widthMm,
    this.heightMm,
    this.wheelbaseMm,
    this.groundClearanceMm,
    this.curbWeightKg,
    this.grossWeightKg,
    this.trunkCapacityLiters,
    this.trunkMaxLiters,
    this.frunkCapacityLiters,
    this.driveType,
    this.frontSuspension,
    this.rearSuspension,
    this.frontBrakes,
    this.rearBrakes,
    this.tireSizeFront,
    this.tireSizeRear,
    this.spareTire,
    this.airbags,
    this.abs,
    this.esc,
    this.tractionControl,
    this.hillStartAssist,
    this.hillDescentControl,
    this.tpms,
    this.isofix,
    this.parkingSensorFront,
    this.parkingSensorRear,
    this.cameraRear,
    this.camera360,
    this.autoParking,
    this.aeb,
    this.fcw,
    this.lka,
    this.ldw,
    this.bsd,
    this.rcta,
    this.acc,
    this.accStopGo,
    this.driverMonitoring,
    this.trafficSignRecognition,
    this.nightVision,
    this.adasLevel,
    this.ncapRating,
    this.ncapBody,
    this.ncapYear,
    this.seats,
    this.seatMaterial,
    this.driverSeatElectric,
    this.passengerSeatElectric,
    this.driverSeatMemory,
    this.ventilatedSeatsFront,
    this.ventilatedSeatsRear,
    this.heatedSeatsFront,
    this.heatedSeatsRear,
    this.rearSeatRecline,
    this.acZones,
    this.rearAcVents,
    this.screenSizeInch,
    this.screenType,
    this.digitalCluster,
    this.clusterSizeInch,
    this.hud,
    this.speakerBrand,
    this.speakerCount,
    this.appleCarplay,
    this.androidAuto,
    this.wirelessCarplay,
    this.wirelessAndroidAuto,
    this.wirelessPhoneCharging,
    this.usbCPorts,
    this.usbAPorts,
    this.bluetooth,
    this.otaUpdate,
    this.headlightType,
    this.drl,
    this.autoHeadlights,
    this.adaptiveHeadlights,
    this.fogLights,
    this.sunroof,
    this.powerTailgate,
    this.handsFreeTailgate,
    this.keylessEntry,
    this.pushStart,
    this.autoFoldingMirrors,
    this.rainSensingWipers,
    this.roofRails,
    this.warrantyYears,
    this.warrantyKm,
    this.batteryWarrantyYears,
    this.batteryWarrantyKm,
  });

  factory CarVariant.fromJson(Map<String, dynamic> json) {
    return CarVariant(
      id: json['id'],
      modelId: json['model_id'],
      name: json['name'],
      priceBaht: (json['price_baht'] as num?)?.toDouble(),
      status: json['status'],
      brandName: json['brand_name'],
      modelName: json['model_name'],
      powertrainType: json['powertrain_type'],
      bodyType: json['body_type'],
      batteryCapacityKwh: (json['battery_capacity_kwh'] as num?)?.toDouble(),
      batteryType: json['battery_type'],
      motorPowerKw: (json['motor_power_kw'] as num?)?.toDouble(),
      motorTorqueNm: (json['motor_torque_nm'] as num?)?.toDouble(),
      frontMotorKw: (json['front_motor_kw'] as num?)?.toDouble(),
      rearMotorKw: (json['rear_motor_kw'] as num?)?.toDouble(),
      rangeKm: json['range_km'],
      rangeStandard: json['range_standard'],
      acChargeKw: (json['ac_charge_kw'] as num?)?.toDouble(),
      dcChargeKw: (json['dc_charge_kw'] as num?)?.toDouble(),
      acChargeTimeHrs: (json['ac_charge_time_hrs'] as num?)?.toDouble(),
      dcChargeTimeMins: json['dc_charge_time_mins'],
      chargingPort: json['charging_port'],
      v2l: json['v2l'],
      v2g: json['v2g'],
      heatPump: json['heat_pump'],
      batteryPreconditioning: json['battery_preconditioning'],
      displacementCc: json['displacement_cc'],
      engineType: json['engine_type'],
      horsepower: json['horsepower'],
      engineTorqueNm: json['engine_torque_nm'],
      fuelType: json['fuel_type'],
      fuelTankLiters: (json['fuel_tank_liters'] as num?)?.toDouble(),
      fuelConsumptionKml: (json['fuel_consumption_kml'] as num?)?.toDouble(),
      turbo: json['turbo'],
      transmission: json['transmission'],
      transmissionSpeeds: json['transmission_speeds'],
      systemPowerHp: json['system_power_hp'],
      systemTorqueNm: json['system_torque_nm'],
      evRangeKm: json['ev_range_km'],
      topSpeedKmh: json['top_speed_kmh'],
      acceleration0100: (json['acceleration_0_100'] as num?)?.toDouble(),
      lengthMm: json['length_mm'],
      widthMm: json['width_mm'],
      heightMm: json['height_mm'],
      wheelbaseMm: json['wheelbase_mm'],
      groundClearanceMm: json['ground_clearance_mm'],
      curbWeightKg: json['curb_weight_kg'],
      grossWeightKg: json['gross_weight_kg'],
      trunkCapacityLiters: json['trunk_capacity_liters'],
      trunkMaxLiters: json['trunk_max_liters'],
      frunkCapacityLiters: json['frunk_capacity_liters'],
      driveType: json['drive_type'],
      frontSuspension: json['front_suspension'],
      rearSuspension: json['rear_suspension'],
      frontBrakes: json['front_brakes'],
      rearBrakes: json['rear_brakes'],
      tireSizeFront: json['tire_size_front'],
      tireSizeRear: json['tire_size_rear'],
      spareTire: json['spare_tire'],
      airbags: json['airbags'],
      abs: json['abs'],
      esc: json['esc'],
      tractionControl: json['traction_control'],
      hillStartAssist: json['hill_start_assist'],
      hillDescentControl: json['hill_descent_control'],
      tpms: json['tpms'],
      isofix: json['isofix'],
      parkingSensorFront: json['parking_sensor_front'],
      parkingSensorRear: json['parking_sensor_rear'],
      cameraRear: json['camera_rear'],
      camera360: json['camera_360'],
      autoParking: json['auto_parking'],
      aeb: json['aeb'],
      fcw: json['fcw'],
      lka: json['lka'],
      ldw: json['ldw'],
      bsd: json['bsd'],
      rcta: json['rcta'],
      acc: json['acc'],
      accStopGo: json['acc_stop_go'],
      driverMonitoring: json['driver_monitoring'],
      trafficSignRecognition: json['traffic_sign_recognition'],
      nightVision: json['night_vision'],
      adasLevel: json['adas_level'],
      ncapRating: (json['ncap_rating'] as num?)?.toDouble(),
      ncapBody: json['ncap_body'],
      ncapYear: json['ncap_year'],
      seats: json['seats'],
      seatMaterial: json['seat_material'],
      driverSeatElectric: json['driver_seat_electric'],
      passengerSeatElectric: json['passenger_seat_electric'],
      driverSeatMemory: json['driver_seat_memory'],
      ventilatedSeatsFront: json['ventilated_seats_front'],
      ventilatedSeatsRear: json['ventilated_seats_rear'],
      heatedSeatsFront: json['heated_seats_front'],
      heatedSeatsRear: json['heated_seats_rear'],
      rearSeatRecline: json['rear_seat_recline'],
      acZones: json['ac_zones'],
      rearAcVents: json['rear_ac_vents'],
      screenSizeInch: (json['screen_size_inch'] as num?)?.toDouble(),
      screenType: json['screen_type'],
      digitalCluster: json['digital_cluster'],
      clusterSizeInch: (json['cluster_size_inch'] as num?)?.toDouble(),
      hud: json['hud'],
      speakerBrand: json['speaker_brand'],
      speakerCount: json['speaker_count'],
      appleCarplay: json['apple_carplay'],
      androidAuto: json['android_auto'],
      wirelessCarplay: json['wireless_carplay'],
      wirelessAndroidAuto: json['wireless_android_auto'],
      wirelessPhoneCharging: json['wireless_phone_charging'],
      usbCPorts: json['usb_c_ports'],
      usbAPorts: json['usb_a_ports'],
      bluetooth: json['bluetooth'],
      otaUpdate: json['ota_update'],
      headlightType: json['headlight_type'],
      drl: json['drl'],
      autoHeadlights: json['auto_headlights'],
      adaptiveHeadlights: json['adaptive_headlights'],
      fogLights: json['fog_lights'],
      sunroof: json['sunroof'],
      powerTailgate: json['power_tailgate'],
      handsFreeTailgate: json['hands_free_tailgate'],
      keylessEntry: json['keyless_entry'],
      pushStart: json['push_start'],
      autoFoldingMirrors: json['auto_folding_mirrors'],
      rainSensingWipers: json['rain_sensing_wipers'],
      roofRails: json['roof_rails'],
      warrantyYears: json['warranty_years'],
      warrantyKm: json['warranty_km'],
      batteryWarrantyYears: json['battery_warranty_years'],
      batteryWarrantyKm: json['battery_warranty_km'],
    );
  }

  String get fullName => '${brandName ?? ''} ${modelName ?? ''} $name'.trim();
}

class CarSearchResult {
  final int variantId;
  final int modelId;
  final String variantName;
  final double? priceBaht;
  final String status;
  final String brandName;
  final String modelName;
  final String powertrainType;
  final int? rangeKm;
  final double? fuelConsumptionKml;

  CarSearchResult({
    required this.variantId,
    required this.modelId,
    required this.variantName,
    this.priceBaht,
    required this.status,
    required this.brandName,
    required this.modelName,
    required this.powertrainType,
    this.rangeKm,
    this.fuelConsumptionKml,
  });

  factory CarSearchResult.fromJson(Map<String, dynamic> json) {
    return CarSearchResult(
      variantId: json['variant_id'],
      modelId: json['model_id'],
      variantName: json['variant_name'],
      priceBaht: (json['price_baht'] as num?)?.toDouble(),
      status: json['status'],
      brandName: json['brand_name'],
      modelName: json['model_name'],
      powertrainType: json['powertrain_type'],
      rangeKm: json['range_km'],
      fuelConsumptionKml: (json['fuel_consumption_kml'] as num?)?.toDouble(),
    );
  }
}
